File: gloSuite9023/gloSuite/gloCommon/gloEmdeonCommon/Forms/frmLab_Graphs.vb
Imports System.Data
Imports System.Data.SqlClient
Imports System.IO
Imports System.Windows.Forms

Public Class frmLab_Graphs
    Inherits System.Windows.Forms.Form

    Private Conn As SqlConnection
    Private Dv As DataView
    Private Cmd As System.Data.SqlClient.SqlCommand
    Dim _flg As Boolean

    Dim _bFlg
    Dim _fromDt
    Dim _ToDt
    Dim _strTest
    Dim _strResults
    Dim gnPatientID As Int64
    Dim gstrMessageBoxCaption As String = ""
    Dim _ConnectionString As String

    '' added by Abhijeet on date 20100514
    Dim appSettings As System.Collections.Specialized.NameValueCollection = System.Configuration.ConfigurationManager.AppSettings

    Public Sub New(ByVal ConnectionString As String, ByVal _PatientID As Int64, Optional ByVal bFlg As Boolean = False, Optional ByVal fromDt As DateTime = Nothing, Optional ByVal ToDt As DateTime = Nothing, Optional ByVal strTest As String = "", Optional ByVal strResults As String = "")

        ' This call is required by the Windows Form Designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.

        Dim sqlconn As String
        sqlconn = ConnectionString
        Conn = New System.Data.SqlClient.SqlConnection(sqlconn)
        'Type = ContactType
        _flg = bFlg
        _fromDt = fromDt
        _ToDt = ToDt
        _strTest = strTest
        _strResults = strResults
        gnPatientID = _PatientID
        _ConnectionString = ConnectionString

        '' Code by : Abhijeet Farkande on date : 20100514
        '' changes : accessing the message box caption from app setting 
        If gstrMessageBoxCaption = "" Then
            If Not IsNothing(appSettings) Then
                If Not IsNothing(appSettings("MessageBOXCaption")) Then
                    If appSettings("") <> "" Then
                        gstrMessageBoxCaption = appSettings("MessageBOXCaption").ToString()
                    Else
                        gstrMessageBoxCaption = "gloEMR"
                    End If
                Else
                    gstrMessageBoxCaption = "gloEMR"
                End If
            Else
                gstrMessageBoxCaption = "gloEMR"
            End If
        End If
        '' End of code by Abhijeet for accessing message box caption     
    End Sub

    Public Function FillControlsTest() As DataTable

        Dim objParamPatientId As SqlParameter = Nothing
        Try
            Dim adpt As New SqlDataAdapter
            Dim dt As New DataTable

            Cmd = New SqlCommand("gsp_Lab_Tests", Conn)
            Cmd.CommandType = CommandType.StoredProcedure

            objParamPatientId = Cmd.Parameters.Add("@nPatientId", SqlDbType.BigInt)
            objParamPatientId.Direction = ParameterDirection.Input
            objParamPatientId.Value = gnPatientID

            adpt.SelectCommand = Cmd
            adpt.Fill(dt)

            Return dt
        Catch ex As SqlClient.SqlException
            Conn.Close()
            '     gloAuditTrail.gloAuditTrail.ExceptionLog(gloAuditTrail.ActivityModule.Labs, gloAuditTrail.ActivityCategory.None, gloAuditTrail.ActivityType.Open, ex.ToString(), gloAuditTrail.ActivityOutCome.Failure)
            '    MessageBox.Show(ex.ToString, gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Error)
            Throw ex
            Return Nothing
        Finally
            If Not IsNothing(objParamPatientId) Then
                objParamPatientId = Nothing
            End If
        End Try
    End Function

    Public Function FillControlsResult() As DataTable
        Dim adpt As New SqlDataAdapter
        Dim dt As New DataTable

        If cmbTests.SelectedItem.Row.ItemArray(1) > 0 Then
            Cmd = New SqlCommand("gsp_Lab_Results", Conn)
            Cmd.CommandType = CommandType.StoredProcedure

            Dim objParamPatientId As SqlParameter
            objParamPatientId = Cmd.Parameters.Add("@nPatientId", SqlDbType.BigInt)
            objParamPatientId.Direction = ParameterDirection.Input
            objParamPatientId.Value = gnPatientID

            objParamPatientId = Cmd.Parameters.Add("@nTest", SqlDbType.BigInt)
            objParamPatientId.Direction = ParameterDirection.Input
            objParamPatientId.Value = cmbTests.SelectedItem.Row.ItemArray(1) '.ValueMember

            adpt.SelectCommand = Cmd
            adpt.Fill(dt)
            Return dt
            If Not IsNothing(objParamPatientId) Then
                objParamPatientId = Nothing
            End If
        End If
        Return Nothing
    End Function

    Private Sub cmbTests_SelectedIndexChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles cmbTests.SelectedIndexChanged
        Try


            If (cmbTests.SelectedItem.Row.ItemArray(1)) > 0 Then

                Dim dt As DataTable
                Dim i As Integer = 0
                dt = FillControlsResult()

                If IsNothing(dt) = False Then
                    cmbResults.DataSource = dt
                    cmbResults.DisplayMember = dt.Columns(0).ColumnName
                    cmbResults.ValueMember = dt.Columns(1).ColumnName
                End If
            End If
        Catch ex As Exception
            gloAuditTrail.gloAuditTrail.ExceptionLog(gloAuditTrail.ActivityModule.Labs, gloAuditTrail.ActivityCategory.None, gloAuditTrail.ActivityType.Query, ex.ToString(), gloAuditTrail.ActivityOutCome.Failure)

        End Try
    End Sub

    Private Sub btnShowGraphs_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ts_btnSave.Click
        Try
            Dim dt_AllPlotValues As New DataTable
            If IsDate(dtFrom.Text) Then
                If IsDate(dtTo.Text) Then
                    If Date.Compare(dtFrom.Value, dtTo.Value) > 0 Then
                        MessageBox.Show("From-date should be less than To-Date ", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information)
                        dtFrom.Focus()
                        Exit Sub
                    End If

                    If Not IsNothing(cmbTests.SelectedValue) Then
                        If Not IsNothing(cmbResults.SelectedValue) Then

                            Dim dtFromdt As DateTime
                            Dim dtTodt As DateTime

                            dtFromdt = CType(dtFrom.Text & " 12:01:00.00 AM", DateTime)
                            dtTodt = CType(dtTo.Text & " 12:01:00.00 AM", DateTime)

                            dt_AllPlotValues = fillData(dtFromdt, dtTodt, , , )

                            Dim dt_MINMAX As New DataTable
                            Dim dt_OnlyMinMax As DataTable

                            '' new data table req. for the data fill for the ranges
                            dt_OnlyMinMax = New DataTable

                            Dim clmnMin As New DataColumn
                            With clmnMin
                                .ColumnName = "Min"
                                '.DataType = System.Type.GetType("System.integer")
                            End With
                            dt_OnlyMinMax.Columns.Add(clmnMin)

                            Dim clmnMax As New DataColumn
                            With clmnMax
                                .ColumnName = "Max"
                                '.DataType = System.Type.GetType("System.Integer")
                                '.DefaultValue = strVal(1)
                            End With
                            dt_OnlyMinMax.Columns.Add(clmnMax)
                            ''''''''

                            Dim j As Integer = 0
                            Dim nActualValue = 0

                            For j = 0 To dt_AllPlotValues.Rows.Count - 1
                                'dt_MINMAX = getMinMAxRanges(dt.Rows(j)(0))  'Split(dt.Rows(j)("labotrd_ResultRange"), "-")
                                'nActualValue = getMinMAxRanges()
                                Dim strVal() As String = Nothing
                                If IsDBNull(dt_AllPlotValues.Rows(j)(0)) = False Then
                                    strVal = Split(dt_AllPlotValues.Rows(j)(0), "-")
                                End If

                                If strVal.Length <= 1 Then
                                    MessageBox.Show("No data available against selected Test ", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information)
                                    Exit Sub
                                End If

                                Dim r As DataRow
                                r = dt_OnlyMinMax.NewRow()
                                If IsNothing(strVal) = False Then
                                    If strVal(0) = "" Then
                                        strVal(1) = "-" & strVal(1)
                                        r.Item(0) = CType(strVal(1), Integer)
                                        r.Item(1) = CType(strVal(2), Integer)
                                    Else
                                        r.Item(0) = CType(strVal(0), Integer)
                                        r.Item(1) = CType(strVal(1), Integer)
                                    End If
                                    'r.Item(0) = CType(strVal(0), Integer)
                                    'r.Item(1) = CType(strVal(1), Integer)
                                    dt_OnlyMinMax.Rows.Add(r)
                                End If
                            Next

                            dt_MINMAX = dt_OnlyMinMax

                            If dt_MINMAX.Rows.Count <= 0 Then
                                MessageBox.Show("No data available against selected Test ", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information)
                                cmbTests.Focus()
                                Exit Sub
                            Else
                                Dim oGraphResult As New frmLab_GraphsResult(_ConnectionString, dtFrom.Text, dtTo.Text, cmbTests.SelectedValue, cmbResults.SelectedValue, gnPatientID, cmbTests.Text, cmbResults.Text, dt_AllPlotValues, dt_OnlyMinMax)
                                With oGraphResult
                                    .MdiParent = Me.Owner
                                    .WindowState = FormWindowState.Maximized
                                    .ShowInTaskbar = False
                                    .BringToFront()
                                    .Show()
                                End With
                                Me.Close()
                            End If
                        Else
                            MessageBox.Show("Please Select Results ", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information)
                            cmbResults.Focus()
                            Exit Sub
                        End If
                    Else
                        MessageBox.Show("Please Select Tests ", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information)
                        cmbTests.Focus()
                        Exit Sub
                    End If
                Else
                    MessageBox.Show("Please Select Valid To Date ", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information)
                    dtTo.Focus()
                    Exit Sub
                End If
            Else
                MessageBox.Show("Please Select Valid From Date ", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information)
                dtFrom.Focus()
                Exit Sub
            End If
        Catch ex As Exception

            gloAuditTrail.gloAuditTrail.ExceptionLog(gloAuditTrail.ActivityModule.Labs, gloAuditTrail.ActivityCategory.None, gloAuditTrail.ActivityType.View, ex.ToString(), gloAuditTrail.ActivityOutCome.Failure)


            MessageBox.Show(ex.Message, gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information)
            Exit Sub
        End Try
    End Sub

    Private Sub frmLab_Graphs_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        Try
            Dim dt As DataTable
            Dim i As Integer = 0

            dt = FillControlsTest()

            dtFrom.Value = System.DateTime.Now
            dtTo.Value = System.DateTime.Now

            If Not IsNothing(dt) Then
                cmbTests.DataSource = dt
                cmbTests.DisplayMember = dt.Columns("TestName").ColumnName
                cmbTests.ValueMember = dt.Columns("TestID").ColumnName
                If dt.Rows.Count > 0 Then
                    cmbTests.SelectedIndex = 0
                End If
            End If

            If _flg = True Then
                If Not IsNothing(_fromDt) Then
                    dtFrom.Value = _fromDt
                End If

                If Not IsNothing(_ToDt) Then
                    dtTo.Value = _ToDt
                End If

                If Not IsNothing(_strTest) Then
                    'Dim testTemp As Integer = 0

                    'For testTemp = 0 To cmbTests.Items.Count - 1
                    '    If cmbTests.Items.Item(testTemp) Then
                    cmbTests.Text = _strTest
                    'Next

                End If

                If Not IsNothing(_strResults) Then
                    cmbResults.Text = _strResults
                End If
            End If

        Catch ex As Exception
            gloAuditTrail.gloAuditTrail.ExceptionLog(gloAuditTrail.ActivityModule.Labs, gloAuditTrail.ActivityCategory.None, gloAuditTrail.ActivityType.Open, ex.ToString(), gloAuditTrail.ActivityOutCome.Failure)

            MessageBox.Show(ex.ToString, gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Error)
        End Try
    End Sub

    Private Sub cmbResults_SelectionChangeCommitted(ByVal sender As Object, ByVal e As System.EventArgs) Handles cmbResults.SelectionChangeCommitted
        ' MessageBox.Show("1." & cmbResults.SelectedValue & "-" & cmbResults.Text & " 2. " & cmbTests.SelectedValue & "-" & cmbTests.Text)
    End Sub

    ' bLab_Flag = false - means call from the same form
    ' bLab_Flag = true -  means call from Lab Order Form

    Public Function fillData(ByVal FrmDate As DateTime, ByVal Todate As DateTime, Optional ByVal bLab_Flag As Boolean = False, Optional ByVal LabTestId As Int64 = 0, Optional ByVal LabResultID As Int64 = 0) As DataTable
        Dim objParamFromDate As SqlParameter = Nothing
        Dim objParamnResultId As SqlParameter = Nothing
        Dim objParamToDate As SqlParameter = Nothing
        Dim objLabFlag As SqlParameter = Nothing
        Dim objParamPatientId As SqlParameter = Nothing
        Dim objParamTestId As SqlParameter = Nothing

        Try
            Dim cmd = New SqlCommand

            Dim oadpt As New SqlDataAdapter
            Dim dt As New DataTable

            cmd = New SqlCommand("gsp_Lab_Graphs", Conn)
            cmd.CommandType = CommandType.StoredProcedure


            objParamPatientId = cmd.Parameters.Add("@nPatientId", SqlDbType.BigInt)
            objParamPatientId.Direction = ParameterDirection.Input
            objParamPatientId.Value = gnPatientID 'gnPatientID '391234534627492001 


            objParamTestId = cmd.Parameters.Add("@nTestId", SqlDbType.BigInt)
            objParamTestId.Direction = ParameterDirection.Input
            If bLab_Flag = False Then
                objParamTestId.Value = cmbTests.SelectedValue   '39221650183922105    '39221650133922105 '39221650133922103 
            Else
                objParamTestId.Value = LabTestId
            End If

            objParamnResultId = cmd.Parameters.Add("@nResultId", SqlDbType.BigInt)
            objParamnResultId.Direction = ParameterDirection.Input
            If bLab_Flag = False Then
                objParamnResultId.Value = cmbResults.SelectedValue  '1
            Else
                objParamnResultId.Value = LabResultID '1
            End If

            ' false = call from same form
            ' true = call from Lab Order form


            objParamFromDate = cmd.Parameters.Add("@dtFromDate", SqlDbType.DateTime)
            objParamFromDate.Direction = ParameterDirection.Input
            If bLab_Flag = False Then
                objParamFromDate.Value = CType(dtFrom.Text & " 12:01:00.00 AM", DateTime)  '"05/21/2007"
            Else
                objParamFromDate.Value = CType(System.DateTime.Today & " 12:01:00.00 AM", DateTime)  '"05/21/2007"
            End If



            objParamToDate = cmd.Parameters.Add("@dtTodate", SqlDbType.DateTime)
            objParamToDate.Direction = ParameterDirection.Input
            If bLab_Flag = False Then
                objParamToDate.Value = CType(dtTo.Text & " 23:59:00.00 PM", DateTime) '"07/29/2007"
            Else
                Todate = Format(Todate, "MM/dd/yyyy")
                objParamToDate.Value = CType(Todate & " 23:59:00.00 PM", DateTime) 'CType(System.DateTime.Today & " 23:59:00.00 PM", DateTime) '"07/29/2007"
            End If



            objLabFlag = cmd.Parameters.Add("@bLabFlg", SqlDbType.Bit)
            objLabFlag.Direction = ParameterDirection.Input
            objLabFlag.Value = bLab_Flag

            oadpt.SelectCommand = cmd
            oadpt.Fill(dt)
            Return dt
        Catch ex As Exception
            'MessageBox.Show(ex.Message, gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Error)
            Throw ex
            Return Nothing
        Finally
            If Not IsNothing(objParamFromDate) Then
                objParamFromDate = Nothing
            End If

            If Not IsNothing(objParamnResultId) Then
                objParamnResultId = Nothing
            End If

            If Not IsNothing(objParamPatientId) Then
                objParamPatientId = Nothing
            End If

            If Not IsNothing(objParamFromDate) Then
                objParamFromDate = Nothing
            End If

            If Not IsNothing(objParamTestId) Then
                objParamTestId = Nothing
            End If

            If Not IsNothing(objParamToDate) Then
                objParamToDate = Nothing
            End If

            If Not IsNothing(objLabFlag) Then
                objLabFlag = Nothing
            End If

        End Try
    End Function

    Private Sub cmbResults_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmbResults.SelectedIndexChanged

    End Sub
End Class

File: gloSuite9023/gloSuite/gloCommon/gloEmdeonInterface/Forms/frmLab_Graphs.cs
using System.Diagnostics;
using System;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.IO;


namespace gloEmdeonInterface.Forms
{
    /// <summary>
    /// Form is not in use right now, this is expected to be convert 
    /// migration functionality of graphs on controls for Test(Results).
    /// </summary>
	public partial class frmLab_Graphs
	{
		
		private SqlConnection Conn;
		//private DataView Dv;--- commented by madan to remove warnings... on 20100520
		
		bool _flg;

        //object _bFlg;--- commented by madan to remove warnings... on 20100520
		object _fromDt;
		object _ToDt;
		object _strTest;
		object _strResults;
        String gstrMessageBoxCaption = "";
        Int64  gnPatientID=0;

        // added by Abhijeet on date 20100417  
        //Below code is commentd by madan on 20100520
        //System.Collections.Specialized.NameValueCollection appSettings = System.Configuration.ConfigurationManager.AppSettings;
        System.Collections.Specialized.NameValueCollection appSettings = System.Configuration.ConfigurationManager.AppSettings;
		
		public frmLab_Graphs(bool bFlg, DateTime fromDt, DateTime ToDt, string strTest, string strResults)
		{
			
			// This call is required by the Windows Form Designer.
			InitializeComponent();
			
			// Add any initialization after the InitializeComponent() call.
			
			string sqlconn;
            sqlconn = "";// GetConnectionString;
			Conn = new System.Data.SqlClient.SqlConnection(sqlconn);
			//Type = ContactType
			_flg = bFlg;
			_fromDt = fromDt;
			_ToDt = ToDt;
			_strTest = strTest;
			_strResults = strResults;

            // Code by : Abhijeet Farkande on date : 20100417, 20100514
            // changes : accessing the message box caption from appSetting               
            if (appSettings != null)
            {                               
                if (appSettings["MessageBOXCaption"] != null)
                {
                    if (appSettings["MessageBOXCaption"] != "")
                    {
                        gstrMessageBoxCaption = Convert.ToString(appSettings["MessageBOXCaption"]);
                    }
                    else
                    {
                        gstrMessageBoxCaption = "gloEMR";
                    }
                }
                else
                { gstrMessageBoxCaption = "gloEMR"; }
            }
            else
            {               
                gstrMessageBoxCaption = "gloEMR";
            }
            // End of code for accessing the user Name                
			
		}
		
		public DataTable FillControlsTest()
		{
		    System.Data.SqlClient.SqlCommand Cmd=null;
            try
            {
                SqlDataAdapter adpt = new SqlDataAdapter();
                DataTable dt = new DataTable();

                Cmd = new SqlCommand("gsp_Lab_Tests", Conn);
                Cmd.CommandType = CommandType.StoredProcedure;

                SqlParameter objParamPatientId;
                objParamPatientId = Cmd.Parameters.Add("@nPatientId", SqlDbType.BigInt);
                objParamPatientId.Direction = ParameterDirection.Input;
                objParamPatientId.Value = gnPatientID;

                adpt.SelectCommand = Cmd;
                adpt.Fill(dt);
                adpt.Dispose();
                adpt = null;
                return dt;
            }
            catch (Exception ex)
            {
                Conn.Close();
                //     gloAuditTrail.gloAuditTrail.ExceptionLog(gloAuditTrail.ActivityModule.Labs, gloAuditTrail.ActivityCategory.None, gloAuditTrail.ActivityType.Open, ex.ToString(), gloAuditTrail.ActivityOutCome.Failure)
                //    MessageBox.Show(ex.ToString, gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Error)
                gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), false);
                return null;
                throw (ex);
                //return null;-- commeted this for fixing warnings... by madan 20100520
            }
            finally
            {
                if (Cmd != null)
                {
                    Cmd.Parameters.Clear();
                    Cmd.Dispose();
                    Cmd = null;
                }
            }
		}
		
		public DataTable FillControlsResult()
		{
            SqlCommand Cmd = null;
            SqlDataAdapter adpt = new SqlDataAdapter();
			DataTable dt = new DataTable();

            if (Convert.ToInt64(cmbTests.SelectedValue) > 0)
			{
				Cmd = new SqlCommand("gsp_Lab_Results", Conn);
				Cmd.CommandType = CommandType.StoredProcedure;
				
				SqlParameter objParamPatientId;
				objParamPatientId = Cmd.Parameters.Add("@nPatientId", SqlDbType.BigInt);
				objParamPatientId.Direction = ParameterDirection.Input;
				objParamPatientId.Value = gnPatientID;
				
				objParamPatientId = Cmd.Parameters.Add("@nTest", SqlDbType.BigInt);
				objParamPatientId.Direction = ParameterDirection.Input;
				objParamPatientId.Value = cmbTests.SelectedValue ; //.ValueMember
				
				adpt.SelectCommand = Cmd;
				adpt.Fill(dt);
                adpt.Dispose();
                adpt = null;
                Cmd.Parameters.Clear();
                Cmd.Dispose();
                Cmd = null;
			}
            return dt;
		}
		
		public void cmbTests_SelectedIndexChanged(object sender, System.EventArgs e)
		{
			try
			{


                if (Convert.ToInt64(cmbTests.SelectedValue) > 0)
				{
					
					DataTable dt;
                    //int i = 0;-- commented this for fixing the warnings.. by madan on 20100520
					dt = FillControlsResult();
					
					if ((dt == null) == false)
					{
						cmbResults.DataSource = dt;
						cmbResults.DisplayMember = dt.Columns[0].ColumnName;
						cmbResults.ValueMember = dt.Columns[1].ColumnName;
					}
				}
			}
			catch (Exception ex)
			{
                gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), false);
				//gloAuditTrail.gloAuditTrail.ExceptionLog(gloAuditTrail.gloAuditTrail.ExceptionLog, gloAuditTrail.gloAuditTrail.ActivityCategory , gloAuditTrail.gloAuditTrail.ExceptionLog, ex.ToString(), gloAuditTrail.gloAuditTrail.ExceptionLog);
				
			}
		}
		
		public void btnShowGraphs_Click(System.Object sender, System.EventArgs e)
		{
            DataTable dt_OnlyMinMax = null;
            try
            {




                if (IsDate(dtFrom.Text))
                {
                    if (IsDate(dtTo.Text))
                    {

                        if (DateTime.Compare(dtFrom.Value, dtTo.Value) > 0)
                        {
                            MessageBox.Show("From-date should be less than To-Date ", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information);
                            dtFrom.Focus();
                            return;
                        }

                        if (cmbTests.SelectedValue != null)
                        {
                            if (cmbResults.SelectedValue != null)
                            {

                                DateTime dtFromdt;
                                DateTime dtTodt;

                                dtFromdt = System.Convert.ToDateTime(dtFrom.Text + " 12:01:00.00 AM");
                                dtTodt = System.Convert.ToDateTime(dtTo.Text + " 12:01:00.00 AM");

                                DataTable dt_AllPlotValues = fillData(dtFromdt, dtTodt, false, 0, 0);

                                //DataTable dt_MINMAX = null;


                                //' new data table req. for the data fill for the ranges
                                dt_OnlyMinMax = new DataTable();

                                DataColumn clmnMin = new DataColumn();
                                clmnMin.ColumnName = "Min";
                                //.DataType = System.Type.GetType("System.integer")
                                dt_OnlyMinMax.Columns.Add(clmnMin);

                                DataColumn clmnMax = new DataColumn();
                                clmnMax.ColumnName = "Max";
                                //.DataType = System.Type.GetType("System.Integer")
                                //.DefaultValue = strVal(1)
                                dt_OnlyMinMax.Columns.Add(clmnMax);
                                ///'''''
                                string[] strVal;
                                int j = 0;
                                //int nActualValue = 0; -- commented to remove warnings... on 20100520 by madan 

                                for (j = 0; j <= dt_AllPlotValues.Rows.Count - 1; j++)
                                {
                                    //dt_MINMAX = getMinMAxRanges(dt.Rows(j)(0))  'Split(dt.Rows(j)("labotrd_ResultRange"), "-")
                                    //nActualValue = getMinMAxRanges()

                                    if ((dt_AllPlotValues.Rows[j][0]) != null)
                                    {
                                        strVal = System.Convert.ToString(dt_AllPlotValues.Rows[j][0]).Split('-');


                                        if (strVal.Length <= 1)
                                        {
                                            MessageBox.Show("No data available against selected Test ", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information);
                                            return;
                                        }
                                        //}
                                        DataRow r;
                                        r = dt_OnlyMinMax.NewRow();
                                        if ((strVal == null) == false)
                                        {
                                            if (strVal[0] == "")
                                            {
                                                strVal[1] = "-" + strVal[1];
                                                r[0] = System.Convert.ToInt32(strVal[1]);
                                                r[1] = System.Convert.ToInt32(strVal[2]);
                                            }
                                            else
                                            {
                                                r[0] = System.Convert.ToInt32(strVal[0]);
                                                r[1] = System.Convert.ToInt32(strVal[1]);
                                            }
                                            //r.Item(0) = CType(strVal(0), Integer)
                                            //r.Item(1) = CType(strVal(1), Integer)
                                            dt_OnlyMinMax.Rows.Add(r);
                                        }
                                    }
                                }
                                if (dt_AllPlotValues != null)
                                {
                                    dt_AllPlotValues.Dispose();
                                    dt_AllPlotValues = null;
                                }

                                //dt_MINMAX = dt_OnlyMinMax;

                                if (dt_OnlyMinMax.Rows.Count <= 0)
                                {
                                    MessageBox.Show("No data available against selected Test ", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information);
                                    cmbTests.Focus();
                                    return;
                                }
                                else
                                {
                                    //object  oGraphResult = new frmLab_GraphsResult(dtFrom.Text, dtTo.Text, cmbTests.SelectedValue, cmbResults.SelectedValue, gnPatientID, cmbTests.Text, cmbResults.Text, dt_AllPlotValues, dt_OnlyMinMax);
                                    //oGraphResult.MdiParent = this.Owner;
                                    //oGraphResult.WindowState = FormWindowState.Maximized;
                                    //oGraphResult.ShowInTaskbar = false;
                                    //oGraphResult.BringToFront();
                                    //oGraphResult.Show();
                                    //this.Close();
                                }
                            }
                            else
                            {
                                MessageBox.Show("Please Select Results ", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information);
                                cmbResults.Focus();
                                return;
                            }
                        }
                        else
                        {
                            MessageBox.Show("Please Select Tests ", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information);
                            cmbTests.Focus();
                            return;
                        }
                    }
                    else
                    {
                        MessageBox.Show("Please Select Valid To Date ", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information);
                        dtTo.Focus();
                        return;
                    }
                }
                else
                {
                    MessageBox.Show("Please Select Valid From Date ", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information);
                    dtFrom.Focus();
                    return;
                }
            }
            catch (Exception ex)
            {

                //gloAuditTrail.gloAuditTrail.ExceptionLog(gloAuditTrail.gloAuditTrail.ExceptionLog, gloAuditTrail.gloAuditTrail.ExceptionLog, gloAuditTrail.gloAuditTrail.ExceptionLog, ex.ToString(), gloAuditTrail.gloAuditTrail.ExceptionLog);
                gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), false);

                MessageBox.Show(ex.Message, gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information);
                return;
            }
            finally
            {
                if (dt_OnlyMinMax != null)
                {
                    dt_OnlyMinMax.Dispose();
                    dt_OnlyMinMax = null;
                }
            }


		}

        private bool IsDate(string testDateTime)
        {
            bool _result = false;

            try
            {
                if (testDateTime.Length > 0)
                {
                    DateTime dt = new DateTime();
                    DateTime.TryParse(dtFrom.Text.Trim(), out dt);
                    _result = true;
                }
                else
                {
                    _result = false;
                }
            }
            catch (Exception ex)
            {
                gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), false);
                _result = false;
            }
            return _result;
        }
		
		private void frmLab_Graphs_Load(object sender, System.EventArgs e)
		{
			try
			{
				DataTable dt;
				//int i = 0; -- commeted this variable to remove warnings by madan on 20100520
				
				dt = FillControlsTest();
				
				dtFrom.Value = System.DateTime.Now;
				dtTo.Value = System.DateTime.Now;
				
				if (dt != null)
				{
					cmbTests.DataSource = dt;
					cmbTests.DisplayMember = dt.Columns["TestName"].ColumnName;
					cmbTests.ValueMember = dt.Columns["TestID"].ColumnName;
					if (dt.Rows.Count > 0)
					{
						cmbTests.SelectedIndex = 0;
					}
				}
				
				if (_flg == true)
				{
					if (_fromDt != null)
					{
						dtFrom.Value =  (System.DateTime) (_fromDt);
					}
					
					if (_ToDt != null)
					{
						dtTo.Value =  (System.DateTime) (_ToDt);
					}
					
					if (_strTest != null)
					{
						//Dim testTemp As Integer = 0
						
						//For testTemp = 0 To cmbTests.Items.Count - 1
						//    If cmbTests.Items.Item(testTemp) Then
						cmbTests.Text = Convert.ToString(_strTest);
						//Next
						
					}
					
					if (_strResults != null)
					{
						cmbResults.Text =  Convert.ToString( _strResults);
					}
				}
				
			}
			catch (Exception ex)
			{
				//gloAuditTrail.gloAuditTrail.ExceptionLog(gloAuditTrail.gloAuditTrail.ExceptionLog, gloAuditTrail.gloAuditTrail.ExceptionLog,, ex.ToString(), gloAuditTrail.gloAuditTrail.ExceptionLog);
                gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), false);
				MessageBox.Show(ex.ToString(), gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Error);
			}
		}
		
		public void cmbResults_SelectionChangeCommitted(object sender, System.EventArgs e)
		{
			// MessageBox.Show("1." & cmbResults.SelectedValue & "-" & cmbResults.Text & " 2. " & cmbTests.SelectedValue & "-" & cmbTests.Text)
		}
		
		// bLab_Flag = false - means call from the same form
		// bLab_Flag = true -  means call from Lab Order Form
		
		public DataTable fillData(DateTime FrmDate, DateTime Todate, bool bLab_Flag, long LabTestId, long LabResultID)
		{
			try
			{
				SqlCommand cmd = new SqlCommand();
				
				SqlDataAdapter oadpt = new SqlDataAdapter();
				DataTable dt = new DataTable();
				
				cmd = new SqlCommand("gsp_Lab_Graphs", Conn);
				cmd.CommandType = CommandType.StoredProcedure;
				
				SqlParameter objParamPatientId;
				objParamPatientId = cmd.Parameters.Add("@nPatientId", SqlDbType.BigInt);
				objParamPatientId.Direction = ParameterDirection.Input;
				objParamPatientId.Value = gnPatientID; //gnPatientID '391234534627492001
				
				SqlParameter objParamTestId;
				objParamTestId = cmd.Parameters.Add("@nTestId", SqlDbType.BigInt);
				objParamTestId.Direction = ParameterDirection.Input;
				if (bLab_Flag == false)
				{
					objParamTestId.Value = cmbTests.SelectedValue; //39221650183922105    '39221650133922105 '39221650133922103
				}
				else
				{
					objParamTestId.Value = LabTestId;
				}
				
				SqlParameter objParamnResultId;
				objParamnResultId = cmd.Parameters.Add("@nResultId", SqlDbType.BigInt);
				objParamnResultId.Direction = ParameterDirection.Input;
				if (bLab_Flag == false)
				{
					objParamnResultId.Value = cmbResults.SelectedValue; //1
				}
				else
				{
					objParamnResultId.Value = LabResultID; //1
				}
				
				// false = call from same form
				// true = call from Lab Order form
				
				SqlParameter objParamFromDate;
				objParamFromDate = cmd.Parameters.Add("@dtFromDate", SqlDbType.DateTime);
				objParamFromDate.Direction = ParameterDirection.Input;
				if (bLab_Flag == false)
				{
					objParamFromDate.Value = System.Convert.ToDateTime(dtFrom.Text + " 12:01:00.00 AM"); //"05/21/2007"
				}
				else
				{
					objParamFromDate.Value = System.DateTime.Today + " 12:01:00.00 AM"; //"05/21/2007"
				}
				
				
				SqlParameter objParamToDate;
				objParamToDate = cmd.Parameters.Add("@dtTodate", SqlDbType.DateTime);
				objParamToDate.Direction = ParameterDirection.Input;
				if (bLab_Flag == false)
				{
					objParamToDate.Value = System.Convert.ToDateTime(dtTo.Text + " 23:59:00.00 PM"); //"07/29/2007"
				}
				else
				{
					Todate = DateTime.Parse(Todate.ToString("MM/dd/yyyy"));
					objParamToDate.Value = Todate + " 23:59:00.00 PM"; //CType(System.DateTime.Today & " 23:59:00.00 PM", DateTime) '"07/29/2007"
				}
				
				
				SqlParameter objLabFlag;
				objLabFlag = cmd.Parameters.Add("@bLabFlg", SqlDbType.Bit);
				objLabFlag.Direction = ParameterDirection.Input;
				objLabFlag.Value = bLab_Flag;
				
				oadpt.SelectCommand = cmd;
				oadpt.Fill(dt);
                oadpt.Dispose();
                oadpt = null;
                cmd.Parameters.Clear();
                cmd.Dispose();
                cmd = null;
				return dt;
			}
			catch (Exception ex)
			{
                gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), false);
				//MessageBox.Show(ex.Message, gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Error)
                return null;
                //throw (ex);
			
			}
		}
		
		public void cmbResults_SelectedIndexChanged(System.Object sender, System.EventArgs e)
		{
			
		}

        private void frmLab_Graphs_Load_1(object sender, EventArgs e)
        {

        }
	}
}
